osmorc/osmorc-jps-plugin/src/org/jetbrains/osgi/jps/build/OsgiBuildSession.java

```diff
@@ -158,6 +158,46 @@ private void doBuild() throws OsgiBuildException {
       }
       mySourceToReport = null;
     }
+    else if (myExtension.isUseBndMavenPlugin()) {
+      File bndFile = myExtension.getBundleDescriptorFile();
+      File base = null;
+      if (bndFile != null && bndFile.isFile()) {
+        // explicitly specified bndfile that exists
+        mySourceToReport = bndFile.getAbsolutePath();
+      }
+      else if (StringUtil.isNotEmpty(myExtension.getBndFileLocation())) {
+        // explicitly specified bndfile that does not exist - fail to report a build error
+        throw new OsgiBuildException(message("session.bnd.missing", myExtension.getBndFileLocation()));
+      }
+      else {
+        // no bndfile specified, try to fallback to the default bnd.bnd file next to the pom.xml
+        File mavenProjectPath = OsgiBuildUtil.getMavenProjectPath(myContext, myModule);
+        if (mavenProjectPath != null) {
```

Comment:
I'd throw a build exception - this is not supposed to happen.

Reply:
Done